jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -316,6 +316,11 @@ else if (filled < 0)
      */
     protected boolean fillAndParseForContent()
     {
+        // Defensive check to avoid an infinite select/wakeup/fillAndParseForContent/wait loop
+        // in case the parser was mistakenly closed and the connection was not aborted.
+        if (_parser.isClose() || _parser.isClosed())
```

Comment:
Didn't we add a `isXxx` that tested both of these states?  If not, let's do so as this double test is done at least one other location.

Reply:
Sure, if @sbordet agrees.

Reply:
@lorban I agree. I don't see the method in 9.4.x, but I would call it `isTerminated()` rather than `isPastEnd()` or similar.
The javadoc for the new method should be enough to explain the state.

Reply:
Okay, I introduced `isTerminated()` by copying the mechanism that was already used to implement `isIdle()` and `isComplete()`.